packages/react/src/components/TextField/TextField.base.tsx

```diff
@@ -248,7 +249,13 @@ export class TextFieldBase extends React.Component<ITextFieldProps, ITextFieldSt
             {iconProps && <Icon className={classNames.icon} {...iconProps} />}
             {hasRevealButton && (
               // Explicitly set type="button" since the default button type within a form is "submit"
-              <button className={classNames.revealButton} onClick={this._onRevealButtonClick} type="button">
+              <button
+                aria-label={revealPasswordAriaLabel}
+                className={classNames.revealButton}
+                onClick={this._onRevealButtonClick}
+                tabIndex={-1}
```

Comment:
```suggestion
                aria-pressed={!!isRevealingPassword}
```
I wouldn't suggest removing this from the tab order -- it would benefit keyboard-only users to be able to access it directly. I'd also suggest adding `aria-pressed` on here to communicate state -- both the current on/off state, and also changes in state when the button is clicked.